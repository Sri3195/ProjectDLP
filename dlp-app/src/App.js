
import './App.css';
import * as React from 'react';
import { Link } from 'react-router-dom';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import digilendImage from './utils/digi-lend.jpg'
import { Card, Grid } from '@mui/material';
import { ArrowForward } from '@mui/icons-material';

//import axios from 'axios';
import { useState } from 'react';
import { useEffect } from 'react';





function App() {
  const [products, setProducts] = useState([])

  const url1 = new URL('http://127.0.0.1:8001/v1/fetch');



  useEffect(() => {

    fetch(url1)
    .then(response=>response.json())
    .then(data => { setProducts(data);
    })
  }, []);
  console.log(products)
  return (
    <>
      <Box position='fixed' sx={{ display: 'flex', alignItems: 'flex-start', flexDirection: 'row' }}>
        <AppBar color='inherit' sx={{ height: 75 }}>
          <Toolbar disableGutters>
            <Typography variant='p' sx={{ ml: 20 }} data-testid="dlp-title" >
              DigiLend
            </Typography>
            <Typography variant='p' sx={{ ml: 30 }} >
              <Link to="/"  data-testid="all-products" style={{ textDecoration: 'none', color: 'black'}}>
                All Products
              </Link>
            </Typography>
            <Typography variant='p' color="black" sx={{ ml: 10 }}>
              <Link to="/"  data-testid="about-us" style={{ textDecoration: 'none', color: 'black' }}>
                About Us
              </Link>
            </Typography>
            <Typography variant='p' color="black" sx={{ ml: 10 }}>
              <Link to="/" data-testid="contact-us" style={{ textDecoration: 'none', color: 'black' }}>
                Contact Us
              </Link>
            </Typography>
            <Button variant='outlined' color='primary' sx={{ ml: 50 }}>SignIn</Button>
          </Toolbar>
        </AppBar>
      </Box>
      <Box>
        <img src={digilendImage} alt="no-image" style={{ marginLeft: '130px', margisnTop: '100px', width: '1250px', height: '400px' }} />
      </Box>
      <Box sx={{ ml: 17, mt: 3 ,mr:10}}>
        <Typography variant='h6' fontWeight='bold' color="black" >
          PRODUCTS
        </Typography>
        <Grid container spacing={6}>
          {products.map(product=>
          <Grid item>
            <Card role='card' sx={{ height: 150, width: 275, mt: 2 ,backgroundColor: '#BADDF9'}}>
              <Typography variant='body1' fontWeight='bold' sx={{ p: 1 }}>{product.productName}</Typography>
              <Typography variant='body2' color='#7B7D83' fontWeight='medium' sx={{ p: 1 }}>{product.productBenefit}</Typography>
              <Box display='flex'>
              <Link to="/phone-number"  state={product.productName} style={{ textDecoration: 'none', color: '#0C0429' }}><Typography variant='body2' sx={{ p: 1 }}>{product.productLink}</Typography></Link>
              <Link to="/phone-number" state={product.productName}style={{ textDecoration: 'none', color: '#0C0429' }}><ArrowForward fontSize='small'sx={{mt:1}}/></Link>
              </Box>
            </Card>
          </Grid>
          )}
        </Grid>
      </Box>
      <Box sx={{ mt:15, backgroundColor: '#7b7d83'}}>
      <Typography variant='p' sx={{ml:50}}>
              DigiLend
            </Typography>
            <Typography variant='p'  sx={{ml:10}}>
              <Link to="/" style={{ textDecoration: 'none', color: 'black' }}>
                All Products
              </Link>
            </Typography>
            <Typography variant='p' color="black" sx={{ml:10}}>
              <Link to="/" style={{ textDecoration: 'none', color: 'black' }}>
                About Us
              </Link>
            </Typography>
            <Typography variant='p' color="black" sx={{ml:10}}>
              <Link to="/" style={{ textDecoration: 'none', color: 'black' }}>
                Contact Us
              </Link>
            </Typography> 
      </Box>

    </>
  );
}

export default App;
