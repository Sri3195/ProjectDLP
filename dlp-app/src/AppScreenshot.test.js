import React from "react";
import { render } from "react-dom";
import {createRenderer} from 'react-test-renderer/shallow'
import App from "./App";

const renderer=createRenderer();

describe('App Screenshot Test',()=>{

    test('renders corretly the app component',()=>{
        renderer.render(<App/>)

        const result=renderer.getRenderOutput();
        expect(result).toMatchSnapshot()
    });



});