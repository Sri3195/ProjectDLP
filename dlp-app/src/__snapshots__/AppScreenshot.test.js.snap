// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Screenshot Test renders corretly 1`] = `
<React.Fragment>
  <ForwardRef(Box)
    position="fixed"
    sx={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <ForwardRef(AppBar)
      color="inherit"
      sx={
        Object {
          "height": 75,
        }
      }
    >
      <ForwardRef(Toolbar)
        disableGutters={true}
      >
        <ForwardRef(Typography)
          data-testid="dlp-title"
          sx={
            Object {
              "ml": 20,
            }
          }
          variant="p"
        >
          DigiLend
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          sx={
            Object {
              "ml": 30,
            }
          }
          variant="p"
        >
          <Link
            data-testid="all-products"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            All Products
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          color="black"
          sx={
            Object {
              "ml": 10,
            }
          }
          variant="p"
        >
          <Link
            data-testid="about-us"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            About Us
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          color="black"
          sx={
            Object {
              "ml": 10,
            }
          }
          variant="p"
        >
          <Link
            data-testid="contact-us"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            Contact Us
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Button)
          color="primary"
          sx={
            Object {
              "ml": 50,
            }
          }
          variant="outlined"
        >
          SignIn
        </ForwardRef(Button)>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
  </ForwardRef(Box)>
  <ForwardRef(Box)>
    <img
      alt="no-image"
      src="digi-lend.jpg"
      style={
        Object {
          "height": "400px",
          "marginLeft": "130px",
          "margisnTop": "100px",
          "width": "1250px",
        }
      }
    />
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "ml": 17,
        "mr": 10,
        "mt": 3,
      }
    }
  >
    <ForwardRef(Typography)
      color="black"
      fontWeight="bold"
      variant="h6"
    >
      PRODUCTS
    </ForwardRef(Typography)>
    <ForwardRef(Grid)
      container={true}
      spacing={6}
    />
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "backgroundColor": "#7b7d83",
        "mt": 15,
      }
    }
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "ml": 50,
        }
      }
      variant="p"
    >
      DigiLend
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        All Products
      </Link>
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="black"
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        About Us
      </Link>
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="black"
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        Contact Us
      </Link>
    </ForwardRef(Typography)>
  </ForwardRef(Box)>
</React.Fragment>
`;

exports[`App Screenshot Test renders corretly the app component 1`] = `
<React.Fragment>
  <ForwardRef(Box)
    position="fixed"
    sx={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <ForwardRef(AppBar)
      color="inherit"
      sx={
        Object {
          "height": 75,
        }
      }
    >
      <ForwardRef(Toolbar)
        disableGutters={true}
      >
        <ForwardRef(Typography)
          data-testid="dlp-title"
          sx={
            Object {
              "ml": 20,
            }
          }
          variant="p"
        >
          DigiLend
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          sx={
            Object {
              "ml": 30,
            }
          }
          variant="p"
        >
          <Link
            data-testid="all-products"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            All Products
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          color="black"
          sx={
            Object {
              "ml": 10,
            }
          }
          variant="p"
        >
          <Link
            data-testid="about-us"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            About Us
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          color="black"
          sx={
            Object {
              "ml": 10,
            }
          }
          variant="p"
        >
          <Link
            data-testid="contact-us"
            style={
              Object {
                "color": "black",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            Contact Us
          </Link>
        </ForwardRef(Typography)>
        <ForwardRef(Button)
          color="primary"
          sx={
            Object {
              "ml": 50,
            }
          }
          variant="outlined"
        >
          SignIn
        </ForwardRef(Button)>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
  </ForwardRef(Box)>
  <ForwardRef(Box)>
    <img
      alt="no-image"
      src="digi-lend.jpg"
      style={
        Object {
          "height": "400px",
          "marginLeft": "130px",
          "margisnTop": "100px",
          "width": "1250px",
        }
      }
    />
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "ml": 17,
        "mr": 10,
        "mt": 3,
      }
    }
  >
    <ForwardRef(Typography)
      color="black"
      fontWeight="bold"
      variant="h6"
    >
      PRODUCTS
    </ForwardRef(Typography)>
    <ForwardRef(Grid)
      container={true}
      spacing={6}
    />
  </ForwardRef(Box)>
  <ForwardRef(Box)
    sx={
      Object {
        "backgroundColor": "#7b7d83",
        "mt": 15,
      }
    }
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "ml": 50,
        }
      }
      variant="p"
    >
      DigiLend
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        All Products
      </Link>
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="black"
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        About Us
      </Link>
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="black"
      sx={
        Object {
          "ml": 10,
        }
      }
      variant="p"
    >
      <Link
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        Contact Us
      </Link>
    </ForwardRef(Typography)>
  </ForwardRef(Box)>
</React.Fragment>
`;
